integration-config/src/test/java/org/symphonyoss/integration/config/LocalIntegrationServiceTest.java

```diff
@@ -58,20 +71,114 @@
 
+  @Spy
+  private ObjectMapper objectMapper;
+
   @Mock
   private Environment environment;
 
   @InjectMocks
-  private IntegrationService service = new LocalIntegrationService();
+  private LocalIntegrationService service;
 
   @Before
   public void init() throws JsonProcessingException {
-    doReturn("mock-configuration.json").when(environment)
+    doReturn(MOCK_CONFIGURATION).when(environment)
         .getProperty(CONFIG_ENV_PROPERTY, DEFAULT_FILE_NAME);
   }
 
+  @After
+  public void finish() {
+    File file = new File(MOCK_CONFIGURATION);
+    if (file.exists()) {
+     file.delete();
+    }
+  }
+
+  @Test(expected = InvalidConfigurationIdException.class)
+  public void testInvalidConfigurationIdException() throws IntegrationConfigException {
+    service.save((IntegrationSettings) null, null);
+  }
+
+  @Test(expected = InvalidInstanceIdException.class)
+  public void testInvalidInstanceIdException() throws IntegrationConfigException {
+    service.save((IntegrationInstance) null, null);
+  }
+
+  @Test(expected = SaveConfigurationException.class)
+  public void testSaveConfigurationException() throws IntegrationConfigException, IOException {
+    doThrow(IOException.class).when(objectMapper).
+        writeValue(any(OutputStream.class), any(IntegrationRepository.class));
+
+    service.init();
+
+    IntegrationSettings jira = service.getIntegrationById("575062074b54ba5e759c0fd9", TEST_USER);
+    jira.setDescription("Integrating new app with Symphony");
+    jira.setEnabled(false);
+    jira.setVisible(true);
+
+    Whitebox.setInternalState(service, "saveFile", true);
+    service.save(jira, TEST_USER);
+  }
+
+  @Test(expected = SaveInstanceException.class)
+  public void testSaveInstanceException() throws IntegrationConfigException, IOException {
+    doThrow(IOException.class).when(objectMapper).
+        writeValue(any(OutputStream.class), any(IntegrationRepository.class));
+
+    service.init();
+
+    IntegrationInstance instance =
+        service.getInstanceById("575062074b54ba5e759c0fd0", "4321", TEST_USER);
+    instance.setCreatorId("new-user");
+    instance.setOptionalProperties(OPTIONAL_PROPERTIES);
+
+    Whitebox.setInternalState(service, "saveFile", true);
+    service.save(instance, TEST_USER);
+  }
+
+  @Test
+  public void testSaveRepositoryLocally() throws IntegrationConfigException, IOException {
+    // Creates a temp file in a temp dir
+    TemporaryFolder tmpDir = new TemporaryFolder();
+    tmpDir.create();
+    String fileName = tmpDir.getRoot().getPath() + "/mock-configuration-tmp.json";
+    File file = new File(fileName);
+    file.createNewFile();
+
+    // Writes mock config values into this temp file
+    InputStream initialStream = getClass().getClassLoader().getResourceAsStream(MOCK_CONFIGURATION);
+    byte[] buffer = new byte[initialStream.available()];
+    initialStream.read(buffer);
+    OutputStream outStream = new FileOutputStream(file);
+    outStream.write(buffer);
```

Comment:
You could perform a file copy using https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#copy(java.nio.file.Path,%20java.nio.file.Path,%20java.nio.file.CopyOption...)